### PR TITLE
Adding Github action to regenerate Lockfile

### PR DESCRIPTION
**Pull request by Hartorn**
## Type of Change

- [ ] 📚 Examples / docs / tutorials / dependencies update
- [ ] 🔧 Bug fix (non-breaking change which fixes an issue)
- [ ] 🥂 Improvement (non-breaking change which improves an existing feature)
- [ ] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🔐 Security fix

## Checklist

- [ ] I've read the [`CODE_OF_CONDUCT.md`](https://github.com/Giskard-AI/ai-inspector/blob/master/CODE_OF_CONDUCT.md) document.
- [ ] I've read the [`CONTRIBUTING.md`](https://github.com/Giskard-AI/ai-inspector/blob/master/CONTRIBUTING.md) guide.
- [ ] I've updated the code style using `make codestyle`.
- [ ] I've written tests for all new methods and classes that I created.
- [ ] I've written the docstring in Google format for all the methods and classes that I used.

### CODE REVIEW

File: .github/workflows/lock-deps.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
+# https://docs.github.com/en/actions/using-workflows/reusing-workflows#calling-a-reusable-workflow
+# https://docs.github.com/fr/actions/using-workflows/workflow-syntax-for-github-actions#exemple--inclusion-de-chemins-dacc%C3%A8s
+
+name: Lock dependencies
+on:
+  pull_request:
```
Comment (andreybavt) [inline .github/workflows/lock-deps.yml:6]: there should be `paths:` pointing to pyproject.toml, no?

File: .github/workflows/lock-deps.yml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+      - name: Configure git
+        run:  |
+          git config --global user.name 'BotLocker'
+          git config --global user.email 'bot.locker@users.noreply.github.com'
+
+      - name: Adding file
+        run:  |
+          git add pdm.lock
+          git commit -m "Regenerating pdm.lock" --allow-empty
```
Comment (andreybavt) [inline .github/workflows/lock-deps.yml:71]: why `--allow-empty` ?

**Review by andreybavt — COMMENTED** [review]
generally fine, I left a couple of comments

File: .github/workflows/lock-deps.yml (hunk 2 above)
Comment (Hartorn) [reply .github/workflows/lock-deps.yml:71]: I feel like it's fine to get an empty commit if the lock is already up to date, to show it's been regenerating either way

**Review by andreybavt — APPROVED** [review]

**Pull request merged**